app/screens/thread_options/options/save_option/index.ts

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import {withDatabase} from '@nozbe/watermelondb/DatabaseProvider';
+import withObservables from '@nozbe/with-observables';
+import {of as of$} from 'rxjs';
+import {switchMap} from 'rxjs/operators';
+
+import {Preferences} from '@constants';
+import {queryPreferencesByCategoryAndName} from '@queries/servers/preference';
+
+import SaveOption from './save_option';
+
+import type {WithDatabaseArgs} from '@typings/database/database';
+
+const enhanced = withObservables(['threadId'], ({threadId, database}: WithDatabaseArgs & {threadId: string}) => {
+    return {
+        isSaved: queryPreferencesByCategoryAndName(database, Preferences.CATEGORY_SAVED_POST, threadId).observe().pipe(
```

Comment:
this is already available from thread_options, same as the other component that needed the post, so you should either have these on the component itself or in thread_options but not in two places.